### PR TITLE
Adapter impinj reader r700

### PR DESCRIPTION
**Pull request by kwennerImpinj**
Adding PnP Bridge Adapter files for R700

Note: in order to build properly for R700 target, need to edit:
.../pnpbridge/deps/azure-iot-sdk-c-pnp/c-utility/CMakeLists.txt  (must enable this option)

    option(use_default_uuid "set use_default_uuid to ON to use the out of the box UUID that comes with the SDK rather than platform specific implementations" ON)

Must also use toolchain file with /build.sh script (add "--toolchain-file" flag), and example toolchain file included in support_files folder.

### CODE REVIEW

**Comment by kwennerImpinj** [comment]
@dipannita-shaw @dipannita08, is there anything else I need to do to get this merged into the adapater-impinj-reader-r700 branch?

File: pnpbridge/src/adapters/src/impinj_reader_r700/helpers/string_manipulation.h
Comment (dipannita08) [inline pnpbridge/src/adapters/src/impinj_reader_r700/helpers/string_manipulation.h:3]: >Str_Trim_Data [](start = 15, length = 13)

Curious why we don't use inbuilt string type here?

File: pnpbridge/src/adapters/src/impinj_reader_r700/impinj_reader_r700.c
Comment (dipannita08) [inline pnpbridge/src/adapters/src/impinj_reader_r700/impinj_reader_r700.c:109]: >ImpinjReader_CreateJsonResponse [](start = 7, length = 31)

Can't we do this with sprintf and a message format?

File: pnpbridge/src/adapters/src/impinj_reader_r700/helpers/string_manipulation.h
Comment (kwennerImpinj) [reply pnpbridge/src/adapters/src/impinj_reader_r700/helpers/string_manipulation.h:3]: Because I'm not familiar with the conventions!  Happy to take any suggestions here.  I guess I was thinking you could save the step of using "strlen()" externally if you just return the length with the string.

File: pnpbridge/src/adapters/src/impinj_reader_r700/support_files/config.json
Comment (dipannita08) [inline pnpbridge/src/adapters/src/impinj_reader_r700/support_files/config.json:27]: Check-in place holders instead of actual values here that may be sensitive.

File: pnpbridge/src/adapters/src/impinj_reader_r700/support_files/config.json
Comment (dipannita08) [inline pnpbridge/src/adapters/src/impinj_reader_r700/support_files/config.json:13]: Insert placeholders like "<insert-id-scope>" to keep sensitive information safe.

File: pnpbridge/src/adapters/src/impinj_reader_r700/impinj_reader_r700.c
Comment (dipannita08) [inline pnpbridge/src/adapters/src/impinj_reader_r700/impinj_reader_r700.c:348]: > SampleEnvironmentalSensor [](start = 49, length = 26)

[nit] Update function name.

File: pnpbridge/src/adapters/src/impinj_reader_r700/impinj_reader_r700.c
Comment (dipannita08) [inline pnpbridge/src/adapters/src/impinj_reader_r700/impinj_reader_r700.c:359]: >IoTHubMessage_Destroy(messageHandle); [](start = 4, length = 37)

Destroy the message handle for each telemetry message handle created inside the while loop

File: pnpbridge/src/adapters/src/impinj_reader_r700/impinj_reader_r700.c
Comment (dipannita08) [inline pnpbridge/src/adapters/src/impinj_reader_r700/impinj_reader_r700.c:323]: >IoTHubMessage_Destroy [](start = 12, length = 21)

Should this message handle ever get created if there is no data in the buffer?

File: pnpbridge/src/adapters/src/impinj_reader_r700/impinj_reader_r700.c
Comment (dipannita08) [inline pnpbridge/src/adapters/src/impinj_reader_r700/impinj_reader_r700.c:616]: >"cmdResponse", res); [](start = 58, length = 20)

From the model it looks like you're expecting a string schema inside this command response. This is probably something that can be cleanly set with a message format. If you expect the adapter to cater to different *types* of schemas for command responses, the _CreateJsonResponse could differentiate the "capability type" by reading component adapter config (This would need to get cached before-hand - possibly during _CreatePnpAdapter) and matched during _OnCommandUpdate for specific components.

**Comment by dipannita08** [comment]
Looks like there are some build errors in curl_wrapper in both [windows ](https://dev.azure.com/busesiot/pnpbridge/_build/results?buildId=303&view=logs&j=ec022e2c-bd5d-5a8d-66fe-055600b3609f&t=ddffd950-9e59-5126-9c76-8b761ed8d5aa&l=1741)and [linux ](https://dev.azure.com/busesiot/pnpbridge/_build/results?buildId=303&view=logs&j=377bee81-b530-5eab-ac40-709ee4d08646&t=07ddd8fe-2f7c-56b7-69b5-aa19833e8340&l=423)(seem to be related to include paths for headers).

**Comment by dipannita08** [comment]
I would suggest adding a short readme with the information on pointing the build script to the toolchain file (with the example you have here), that way anybody trying to test it out can build it without issues.

**Comment by dipannita08** [comment]
I'm not sure why you would need the uuid dependency since you don't seem to be using any uuid utilities - I'm curious what kind of build errors were you seeing without the cmake change in the deps folder? In any case, did you pull all the submodule dependencies from master?

**Comment by kwennerImpinj** [comment]
@dipannita08 thank you for reviewing these!  I will fix the things that I can.  However, I likely cannot fix all of these before merging.  My thought here was to simply get the files into this branch so that I can continue development from here.  I know they will not be ready for public consumption until later, but I do plan to prepare them before merging to master.  Can we accept a broken merge since it is into a branch an not master?

**Comment by dipannita08** [comment]
> @dipannita08 thank you for reviewing these! I will fix the things that I can. However, I likely cannot fix all of these before merging. My thought here was to simply get the files into this branch so that I can continue development from here. I know they will not be ready for public consumption until later, but I do plan to prepare them before merging to master. Can we accept a broken merge since it is into a branch and not master?

Hey Kevin, no problem, I'm okay with merging this change into this branch and continuing development. Let's come back to these comments when merging to master.

**Review by dipannita08 — APPROVED** [review]
Approving this PR to unblock merging into pnpbridgedev-impinj. We can refer to this PR for comments when merging into master!

**Comment by kwennerImpinj** [comment]
> > @dipannita08 thank you for reviewing these! I will fix the things that I can. However, I likely cannot fix all of these before merging. My thought here was to simply get the files into this branch so that I can continue development from here. I know they will not be ready for public consumption until later, but I do plan to prepare them before merging to master. Can we accept a broken merge since it is into a branch and not master?
> 
> Hey Kevin, no problem, I'm okay with merging this change into this branch and continuing development. Let's come back to these comments when merging to master.

Sounds good @dipannita08 .  Anything I need to do from my end to complete the merge to this dev branch?  Or is that triggered from your side?  I don't see a way to do it myself.

**Comment by kwennerImpinj** [comment]
> > > @dipannita08 thank you for reviewing these! I will fix the things that I can. However, I likely cannot fix all of these before merging. My thought here was to simply get the files into this branch so that I can continue development from here. I know they will not be ready for public consumption until later, but I do plan to prepare them before merging to master. Can we accept a broken merge since it is into a branch and not master?
> > 
> > 
> > Hey Kevin, no problem, I'm okay with merging this change into this branch and continuing development. Let's come back to these comments when merging to master.
> 
> Sounds good @dipannita08 . Anything I need to do from my end to complete the merge to this dev branch? Or is that triggered from your side? I don't see a way to do it myself.

@dipannita08 , looks like @Shyamal-V needs to approve also?

**Comment by usivagna** [comment]
Merging the changes, per @dipannita08  comments, we will re-visit necessary fixes when the code is merged into master.

**Comment by kwennerImpinj** [comment]
Updated the source branch to remove sensitive information in the config.json file.  Ready to merge?

**Comment by dipannita08** [comment]
Squashing and merging into the impinj-dev branch. Let's keep the PR open to review comments later on.

File: pnpbridge/src/adapters/src/impinj_reader_r700/helpers/string_manipulation.h
Comment (kwennerImpinj) [reply pnpbridge/src/adapters/src/impinj_reader_r700/helpers/string_manipulation.h:3]: fixed in a508473

File: pnpbridge/src/adapters/src/impinj_reader_r700/impinj_reader_r700.c
Comment (kwennerImpinj) [reply pnpbridge/src/adapters/src/impinj_reader_r700/impinj_reader_r700.c:109]: Yes i believe we can!  fixed in - a96d8a2: replaced strcat with sprintf commands

File: pnpbridge/src/adapters/src/impinj_reader_r700/support_files/config.json
Comment (kwennerImpinj) [reply pnpbridge/src/adapters/src/impinj_reader_r700/support_files/config.json]: resolved before PR #52 merged

**Pull request merged**
